src/sentry/grouping/strategies/base.py

```diff
@@ -132,20 +169,22 @@ def __init__(
         name: str,
         interface: str,
         score: Optional[int],
-        func: StrategyFunc,
+        func: "StrategyFunc[ConcreteInterface]",
     ):
         self.id = id
         self.strategy_class = id.split(":", 1)[0]
         self.name = name
         self.interface = interface
         self.score = score
         self.func = func
-        self.variant_processor_func: Optional[StrategyFunc] = None
+        self.variant_processor_func: Optional["VariantProcessor"] = None
```

Comment:
Those strings are there because we defined all of our types conditionally.

Reply:
Once 3.8 hits we can do `from __future__ import annotations` and the strings won't be necessary.

Reply:
with 3.8 I think all of the features we currently import from typing_extensions can be used